internal/DrumPicker/DrumPicker.js

```diff
@@ -0,0 +1,783 @@
+import {is} from '@enact/core/keymap';
+import {clamp} from '@enact/core/util';
+import Spottable from '@enact/spotlight/Spottable';
+import IdProvider from '@enact/ui/internal/IdProvider';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import {Children, Component} from 'react';
+import ReactDOM from 'react-dom';
+
+import $L from '../$L';
+import Skinnable from '../../Skinnable';
+
+import DrumPickerItem from './DrumPickerItem';
+
+import css from './DrumPicker.module.less';
+import * as ri from '@enact/ui/resolution';
+
+const Div = Spottable('div');
+
+// Set-up event forwarding
+const forwardKeyDown = forward('onKeyDown');
+
+const isDown = is('down');
+const isLeft = is('left');
+const isRight = is('right');
+const isUp = is('up');
+
+/**
+ * The base component for {@link agate/internal/DrumPicker.DrumPicker}.
+ *
+ * @class DrumPicker
+ * @memberof agate/internal/DrumPicker
+ * @ui
+ * @private
+ */
+const DrumPickerBase = class extends Component {
+	static displayName = 'DrumPicker';
+
+	static propTypes = /** @lends agate/internal/DrumPicker.DrumPicker.prototype */ {
+		/**
+		 * The maximum value selectable by the picker (inclusive).
+		 *
+		 * The range between `min` and `max` should be evenly divisible by
+		 * [step]{@link agate/internal/DrumPicker.DrumPicker.step}.
+		 *
+		 * @type {Number}
+		 * @required
+		 * @public
+		 */
+		max: PropTypes.number.isRequired,
+
+		/**
+		 * The minimum value selectable by the picker (inclusive).
+		 *
+		 * The range between `min` and `max` should be evenly divisible by
+		 * [step]{@link agate/internal/DrumPicker.DrumPicker.step}.
+		 *
+		 * @type {Number}
+		 * @required
+		 * @public
+		 */
+		min: PropTypes.number.isRequired,
+
+		/**
+		 * Accessibility hint
+		 *
+		 * For example, `hour`, `year`, and `meridiem`
+		 *
+		 * @type {String}
+		 * @default ''
+		 * @public
+		 */
+		accessibilityHint: PropTypes.string,
+
+		/**
+		 * The "aria-label" for the picker.
+		 *
+		 * By default, `aria-valuetext` is set to the current value.
+		 * This should only be used when the parent controls the value of
+		 * the picker directly through the props.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		'aria-label': PropTypes.string,
+
+		/**
+		 * Overrides the `aria-valuetext` for the picker. By default, `aria-valuetext` is set
+		 * to the current value. This should only be used when the parent controls the value of
+		 * the picker directly through the props.
+		 *
+		 * @type {String|Number}
+		 * @memberof agate/internal/DrumPicker.DrumPicker.prototype
+		 * @public
+		 */
+		'aria-valuetext': PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+		/**
+		 * Children from which to pick
+		 *
+		 * @type {Node}
+		 * @public
+		 */
+		children: PropTypes.node,
+
+		/**
+		 * Class name for component.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		className: PropTypes.string,
+
+		/**
+		 * Customize component style
```

Comment:
Suggested change: " * Customizes the component by mapping the supplied collection of CSS class names to the
		 * corresponding internal elements and states of this component."